src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -554,7 +555,7 @@ char* java_lang_String::as_quoted_ascii(oop java_string) {
   if (length == 0) return nullptr;
```

Comment:
Should this be checking for length <= 0?  It looks like length can indeed be negative if UTF8::unicode_length() tries to return the length of a utf8 string with length > 0x7fffffff.

Reply:
??? length is assigned on line 552 and again comes from the length of a Java array.

Reply:
OK.